Jenkinsfile

```diff
@@ -1,19 +1,31 @@
 elifeLibrary {
     def commit
+
     stage 'Checkout', {
         checkout scm
         commit = elifeGitRevision()
     }
 
-    stage 'Build image', {
-        sh './build_container.sh'
-    }
+    node('containers-jenkins-plugin') {
+        stage 'Build images', {
+            checkout scm
+            dockerComposeBuild(commit)
+        }
 
-    stage 'Run tests', {
-        elifeLocalTests './project_tests.sh'
+        stage 'Project tests', {
+            dockerComposeRun(
+                "sciencebeam",
+                "./project_tests.sh",
+                commit
+            )
+        }
     }
 
     elifeMainlineOnly {
+        stage 'Merge to master', {
+            elifeGitMoveToBranch commit, 'master'
+        }
+
         stage 'Push image', {
             def image = DockerImage.elifesciences(this, 'sciencebeam', 'latest')
```

Comment:
That might need to change.. how?

Reply:
See https://github.com/elifesciences/annotations/blob/971b71b5e97a97a2a5af3ff61ffb04611abc4100/Jenkinsfile#L29

Reply:
Will that push also _latest_? (since we don't have a released versions for sciencebeam yet)

Reply:
No, will only push a tag with the same value as the commit.

[This will push a named tag](https://github.com/elifesciences/annotations/blob/971b71b5e97a97a2a5af3ff61ffb04611abc4100/Jenkinsfile#L57) named `approved` (or anything else you like). The `.pull()` additional call is to pull the commit-tagged image in case the node this is running on is different from the node that created it as each `node() { ... }` block can be running in a different place.

Reply:
i.e. in the current code just swap 'latest' and commit around?